pkg/util/binfetcher/binfetcher.go

```diff
@@ -44,20 +54,154 @@ type Options struct {
 	// These are optional and rarely need to be set.
 	Component string  // empty for auto-inferred; this is the "loadgen" in "loadgen/kv"
 	Suffix    string  // empty for auto-inferred; either empty, ".zip", or ".tgz"
-	URL       url.URL // empty for auto-inferred
+	URL       url.URL // empty for auto-inferred; ignored for script
+
+	script []string
+}
+
+// Generated outputs a shell script that implements binary fetching. Panics on error.
+func (opts Options) Generated() string {
+	if err := opts.init(); err != nil {
+		panic(err)
+	}
+
+	s := fmt.Sprintf(`#!/usr/bin/env bash
+set -euo pipefail
+
+# A POSIX variable
+# Reset in case getopts has been used previously in the shell.
+OPTIND=1
+
+binary="%s"
+version="%s"
+os=""
+arch="amd64" # TODO(tschottdorf): how to auto-detect?
```

Comment:
`$(uname -m)`

Reply:
However since these scripts are invoked from Go code I highly recommend you to pass the arch as argument from Go (you probably want the same arch as the cockroachdb binary, i.e. the string from the `build` package.

Reply:
The script shouldn't be tied to the arch on which it was generated. The idea is that this can be used in CI for various auxiliary components that need to run a cluster of a certain version (if I understood @benesch correctly). So it wants to just "get the right one for the system on which the script is called".

Also, on my MacBook `uname -m` results in `x86_64`.

Reply:
oh that's unfortunate.

My personal preference would be to name the binaries after the output of `uname -m` on each platform (because that's stable and reliable!) not what Go thinks it should be (that's usually... ad-hoc, if not to say bogus).

Oh well.

However my other proposal remains -- compute the string in Go and bake it in the script by string substitution?

Reply:
What about `go env GOARCH`?

Reply:
mmm I guess nothing else in this script depends on Go being available. Never mind.